src/bulls_and_cows_lib/game_solver.cpp

```diff
@@ -1,7 +1,100 @@
 
 #include "game_solver.hpp"
+#include "random.hpp"
 
 namespace bulls_and_cows {
-    // TODO: define the body of the functions declared in game_solver.cpp
 
-} // namespace bulls_and_cows
+     // TODO: define the body of the functions declared in game_solver.cpp
+
+    void recursif(int num, int max, const GameOptions& game_options, PossibleSolutions& pls, std::string used_alphabeat,
+                  Code codes)
+    {
+        for (char temp : used_alphabeat)
+        {
+            if (num <= max)
+            {
+                codes.value.push_back(temp);
+                num++;
+                recursif(num, max, game_options, pls, used_alphabeat, codes);
```

Comment:
pourquoi fais-tu le pop_back ici ? (je sais pouquoi tu le fais, mais vois-tu un pb avec la façon dont tu as déclaré ton parametre)

Reply:
En fait, je viens de voir que le pop_back était nécessaire, donc tu peux ignorer mon commentaire précédent

Reply:
J'ai rajouté un potit Code& de manière à toujours pointer sur le même élément et ainsi ne pas créer une copie à chaque itération récursive. Est-ce que  c'est ca le pb même si ca fonctionnait sans ?
void generation_all_codes(int num, int max, const GameOptions& game_options, PossibleSolutions& pls, std::string used_alphabeat, Code& codes)

Reply:
Oui, c'est + efficace.
Le problème dont je parlais était un faux problème, oublie :-) (j'avais mal compris ce que faisait ton code :-))